src/R/population_line_plot_comparative.R

```diff
@@ -0,0 +1,80 @@
+population_count_plot_comparative = function(baseline_filenameandpath, scenario_filenameandpath){
+  
+  # DATA
+  
+  ## historic data
+  
+  ### import historic data - taken from send_charts.R
+  remove_colons <-  function(x) {str_replace(x, ':', '')}
+  df_historical <- read.csv("target/historic-data.csv") %>%
+    mutate_all(funs(remove_colons)) %>%
+    filter(need.1 != "NONSEND")
+  
+  n_hist_years <-  df_historical %>%
+    summarise(n_distinct(calendar.year)) + 1.5
+  
+  ### extract year and population counts
+  count_data_historic = df_historical %>%
+    group_by(calendar.year) %>%
+    count() %>%
+    rename(mean=n, calendar.year.str=calendar.year) %>%
+    mutate(calendar.year =  as.numeric(calendar.year.str))
+  
+  
+  ## projections
+  
+  ### import the baseline dataset
+  count_data_baseline = read.csv(baseline_filenameandpath) %>%
+    mutate(group = "baseline")
+  
+  ### import the scenario dataset
+  count_data_scenario = read.csv(scenario_filenameandpath) %>%
+    mutate(group = "scenario") %>%
+    bind_rows(count_data_historic)
+  
+  
+  # PLOTS
+  
+  # ribbon plot version - not zero-indexed
+  g <- ggplot(count_data_baseline, aes(x=calendar.year)) +
+    geom_line(aes(y=mean), colour=cols[1], alpha = alpha_line) +
+    geom_ribbon(aes(ymin=q1, ymax=q3), fill=cols[1], alpha = alpha_ribbon) +
+    geom_line(data=count_data_scenario, aes(y=mean), colour=cols[2], alpha = alpha_line) +
+    geom_ribbon(data=count_data_scenario, aes(ymin=q1, ymax=q3), fill=cols[2], alpha = alpha_ribbon) +
+    labs(y="Total SEND Population") +
+    scale_x_continuous(name="Year",
+                       breaks = seq(min(count_data_historic$calendar.year), max(count_data_baseline$calendar.year)),
+                       limits = c(min(count_data_historic$calendar.year), max(count_data_baseline$calendar.year)+1.5)) +
+    ggtitle("SEND Population Projection") +
+    theme(axis.text.x = element_text(size=8)) +
+    geom_vline(xintercept = max(count_data_historic$calendar.year) + 0.5,
+               color = "dodgerblue",
+               linetype = "dashed")  +
+    geom_text(x=max(count_data_historic$calendar.year) + 0.5,
+              y=max(count_data_baseline$q3, na.rm=T),
+              label = "<-- Historical      Projected -->",
+              color = "dodgerblue") +
+    geom_text(x=2027+1,
```

Comment:
Is it possible to pull the 2027 from the data, rather than hardcode it? We might not always end projections there.

Reply:
done